source/fe/fe_values.cc

```diff
@@ -4788,6 +4788,23 @@ FEFaceValues<dim, spacedim>::reinit(
 
 
 
+template <int dim, int spacedim>
+template <template <int, int> class DoFHandlerType, bool lda>
+void
+FEFaceValues<dim, spacedim>::reinit(
+  const TriaIterator<DoFCellAccessor<DoFHandlerType<dim, spacedim>, lda>> &cell,
+  const TriaIterator<TriaAccessor<dim - 1, dim, spacedim>> &               face)
+{
+  const auto faces   = cell->face_iterators();
+  const auto face_it = std::find(faces.begin(), faces.end(), face);
+  Assert(face_it != faces.end(),
+         ExcMessage("The provided face iterator does not correspond to a face "
+                    "owned by the provided cell."));
+  reinit(cell, face_it - faces.begin());
```

Comment:
This requires more work than I think is necessary as it builds all N face iterators even though on average you'll only need to check N/2. It would be easier if you just wrote
```
  for (unsigned int f=0; f<GeometryInfo<dim>::faces_per_cell; ++f)
    if (cell->face(f) == face)
      {
         reinit (cell, f);
         return;
      }

  Assert (false, ...);
```
Same for the other functions below.